platform/entanglement/src/services/LinkService.js

```diff
@@ -39,42 +39,28 @@ define(
         }
 
         LinkService.prototype.validate = function (object, parentCandidate) {
-            if (!parentCandidate || !parentCandidate.getId) {
-                return false;
-            }
-            if (parentCandidate.getId() === object.getId()) {
-                return false;
-            }
-            if (parentCandidate.getModel().composition.indexOf(object.getId()) !== -1) {
-                return false;
-            }
-            return this.policyService.allow(
-                "composition",
-                parentCandidate.getCapability('type'),
-                object.getCapability('type')
-            );
+            var objectId = object.getId();
```

Comment:
Would love to have a guide for style here, because this looks like a lot of changes when it's really not a lot of changes.  This also becomes consistent here but inconsistent with the rest of the bundle.  Also, boolean chaining with && causes adding/removing of criteria to create more verbose diffs than necessary.  Personally, a 9 line return statement chained with && is a total mess.

Re JSDOC; If link/move/copy have different validation logic they should have different descriptions.  Seems we've lost some detail in our documentation due to the "AbstractComposeService", which itself is pretty unclear.